core/src/main/java/com/scalar/db/storage/jdbc/query/SimpleSelectQuery.java

```diff
@@ -50,21 +55,32 @@ public class SimpleSelectQuery implements SelectQuery {
     isRangeQuery = builder.isRangeQuery;
     indexedColumn = builder.indexedColumn;
     isConditionalQuery = builder.isConditionalQuery;
+    isRelationalQuery = builder.isRelationalQuery;
+    conjunctions = builder.conjunctions;
   }
 
   @Override
   public String sql() {
-    String sql =
-        "SELECT "
-            + projectionSqlString()
-            + " FROM "
-            + rdbEngine.encloseFullTableName(schema, table);
-    if (isConditionalQuery) {
-      sql += " WHERE " + conditionSqlString();
+    StringBuilder builder =
+        new StringBuilder(
+            "SELECT "
+                + projectionSqlString()
+                + " FROM "
+                + rdbEngine.encloseFullTableName(schema, table));
```

Comment:
```suggestion
        new StringBuilder( "SELECT ")
        .append(projectionSqlString())
        .append(" FROM ")
        .append(rdbEngine.encloseFullTableName(schema, table));
```
We could use the builder for this concatenation.

Reply:
JFYI : String concatenation in a single line is implicitly converted into StringBuilder as a part of optimization in recent Java versions.

```java
public class Test {
  public static void main(String[] args) {
    String s = "AAAA " + System.currentTimeMillis() + " BBBB";
  }
}
```

For instance, this code is compiled into the following class file

```
$ javap -c Test
Warning: Binary file Test contains org.example.Test
Compiled from "Test.java"
public class org.example.Test {
  public org.example.Test();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method java/lang/Object."<init>":()V
       4: return

  public static void main(java.lang.String[]);
    Code:
       0: new           #2                  // class java/lang/StringBuilder
       3: dup
       4: invokespecial #3                  // Method java/lang/StringBuilder."<init>":()V
       7: ldc           #4                  // String AAAA
       9: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      12: invokestatic  #6                  // Method java/lang/System.currentTimeMillis:()J
      15: invokevirtual #7                  // Method java/lang/StringBuilder.append:(J)Ljava/lang/StringBuilder;
      18: ldc           #8                  // String  BBBB
      20: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      23: invokevirtual #9                  // Method java/lang/StringBuilder.toString:()Ljava/lang/String;
      26: astore_1
      27: return
}
```

And you can see the optimization. But it's not easy to always confirm the optimization is performed, so explicitly using StringBuilder may be a good option.

Reply:
I see. Thank you for going into detail about this. I did not know about this tool, it is quite handy.
 
My concern was that with an example that somewhat mirrors the current code.
```java
 public static void main(String[] args) {
    StringBuilder builder = new StringBuilder("AAAA " + System.currentTimeMillis() + " BBBB");
    if (System.currentTimeMillis() % 2 == 0) {
      builder.append("CCCC");
    } else {
      builder.append("DDDD");
    }
    String s = builder.toString();
  }
```
A first StringBuilder is created to concatenate `"AAAA "+ System.currentTimeMillis()+" BBBB"` and then a second is created to append the string in the `if` condition instead of reusing the first one. 
```
public class com.scalar.db.api.StrBuild {
  public com.scalar.db.api.StrBuild();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method java/lang/Object."<init>":()V
       4: return

  public static void main(java.lang.String[]);
    Code:
       0: new           #2                  // class java/lang/StringBuilder
       3: dup
       4: new           #2                  // class java/lang/StringBuilder
       7: dup
       8: invokespecial #3                  // Method java/lang/StringBuilder."<init>":()V
      11: ldc           #4                  // String AAAA
      13: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      16: invokestatic  #6                  // Method java/lang/System.currentTimeMillis:()J
      19: invokevirtual #7                  // Method java/lang/StringBuilder.append:(J)Ljava/lang/StringBuilder;
      22: ldc           #8                  // String  BBBB
      24: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      27: invokevirtual #9                  // Method java/lang/StringBuilder.toString:()Ljava/lang/String;
      30: invokespecial #10                 // Method java/lang/StringBuilder."<init>":(Ljava/lang/String;)V
      33: astore_1
      34: invokestatic  #6                  // Method java/lang/System.currentTimeMillis:()J
      37: ldc2_w        #11                 // long 2l
      40: lrem
      41: lconst_0
      42: lcmp
      43: ifne          56
      46: aload_1
      47: ldc           #13                 // String CCCC
      49: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      52: pop
      53: goto          63
      56: aload_1
      57: ldc           #14                 // String DDDD
      59: invokevirtual #5                  // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      62: pop
      63: aload_1
      64: invokevirtual #9                  // Method java/lang/StringBuilder.toString:()Ljava/lang/String;
      67: astore_2
      68: return
}
```
As you say, a more recent compiler might optimize that case better, but since we refactored to use a StringBuilder on purpose in this method, we could use it for all of the concatenations.

Reply:
Thank you for the information. I agree it's better to use append here. Fixed in 4b2fbe5.